src/Microsoft.AspNet.OData/Adapters/WebApiRequestMessage.cs

```diff
@@ -26,7 +26,7 @@ internal class WebApiRequestMessage : IWebApiRequestMessage
         /// <summary>
         /// The inner request wrapped by this instance.
         /// </summary>
-        private HttpRequestMessage innerRequest;
+        internal HttpRequestMessage innerRequest;
```

Comment:
Looking at the general design of the class, we do not need to change this field from `private` to `internal`. Seems the original intention was to disallow changing the `HttpRequestMessage` object once created. I would prefer to add another method/property in the class `WebApiRequestMessage` as follows and use that to get configurations.

```
        public HttpConfiguration Configuration
        {
            get { return this.innerRequest.GetConfiguration(); }
        }
```

Reply:
Great suggestion. However the NetCore `WebApiRequestMessage` class is marked as `internal`. I thought it would be consistent to make the NetClassic version `internal` too. https://github.com/OData/WebApi/blob/master/src/Microsoft.AspNetCore.OData/Adapters/WebApiRequestMessage.cs

Reply:
#Resolved